tools/CheckDates.scala

```diff
@@ -0,0 +1,110 @@
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change contains "Documents date fixed" it will be ignored and 
+    we will take next commit.
+
+    analyzeDates - just shows if anything is ok
+    fixWrongDates - actual fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyzeDates
+           scala tools/CheckDates.scala fixWrongDates
+
+  """
+
+  def readFilesList(): Array[String] = {
+    val myval =  ("find . -name *.scala ! -path ./tools/*  -type f") !!
+    val files_array = myval.split("\n").map(_.toString).distinct
+    
+    return files_array
+  }
+
+  def readFile(pathToFile: String): String = {
+    val source = scala.io.Source.fromFile(pathToFile)
+    val lines = try source.mkString finally source.close()
+
+    return lines
+  }
+
+  def githubDate(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val messages = buf_messages.split("commit").distinct.drop(1)
+    val dates = buf_dates.split("\\n").distinct
+    var gits_info = Array[(String, String)]()
+    for (a <- 0 to dates.length - 1) {
+      gits_info = gits_info :+ (messages(a),dates(a))
+    }
+    val res = gits_info.filter(!_._1.contains("Documents date fixed"))
+
+    return res(0)._2
+  }
+
+  def ifFileWasInGit(pathToFile: String): Boolean = {
+    val res_dates = (s"git log -3 $pathToFile" !!)
+    if(res_dates == "") {
+      return false
+    } else {
+      return true
+    }
+  }
+
+  def analyzeDates(): Array[(String, String, String)] = {
+    val files_array = readFilesList()
+    var data = Array[(String, String, String)]()
+    for(file_path <- files_array) {
+      if(readFile(file_path).contains("(c) 2003-") && ifFileWasInGit(file_path)) {
+        val file = readFile(file_path)
+        val start_index = file.indexOf("(c) 2003-")
+        val actualYear = file.substring(start_index + 9, start_index + 13)
+        val gitYear = githubDate(file_path).split(" ")(4)
+        data = data :+ (file_path, actualYear, gitYear) 
+      }
+    }
+
+    return data
+  }
+
+  def fixWrongDates(): Unit = {
+    val files = analyzeDates()
+    files.foreach {file =>
+      if (file._2.toInt != file._3.toInt) {
+        val actual_file = readFile(file._1)
+        val new_file = actual_file.replace(s"(c) 2003-${file._2}", s"(c) 2003-${file._3}")
+        scala.tools.nsc.io.File(file._1).writeAll(new_file)
+      }
+    }
+  }
+
+  def resultAnalyzeDates(): Unit = {
+    val data = analyzeDates
+    var bool = false
+    data.foreach {file =>  
+      if (file._2.toInt != file._3.toInt) {
+        println(file + " wrong date ")
+        bool = true
+      } else {
+        println(file + " ok ")
+      }
+    }
+    if(bool) println("Oops...We have some error dates")
+    else println ("Everything is ok")
+  }
+
+  def main(args: Array[String]) {
```

Comment:
Extend App?
```
The argument of the scala command has to be a top-level object. If that object extends trait App, then all statements contained in that object will be executed; otherwise you have to add a method main which will act as the entry point of your program.
```

I choosed `main`, so now I don't see a point in `Extend App`. Am I wrong?